src/main/java/it/gov/pagopa/pu/debtpositions/service/create/debtposition/CreateDebtPositionServiceImpl.java

```diff
@@ -0,0 +1,70 @@
+  private final DebtPositionService debtPositionService;
+  private final GenerateIuvService generateIuvService;
+  private final InstallmentNoPIIRepository installmentNoPIIRepository;
+
+  public CreateDebtPositionServiceImpl(AuthorizeOperatorOnDebtPositionTypeService authorizeOperatorOnDebtPositionTypeService,
+                                       ValidateDebtPositionService validateDebtPositionService,
+                                       DebtPositionService debtPositionService, GenerateIuvService generateIuvService,
+                                       InstallmentNoPIIRepository installmentNoPIIRepository) {
+    this.authorizeOperatorOnDebtPositionTypeService = authorizeOperatorOnDebtPositionTypeService;
+    this.validateDebtPositionService = validateDebtPositionService;
+    this.debtPositionService = debtPositionService;
+    this.generateIuvService = generateIuvService;
+    this.installmentNoPIIRepository = installmentNoPIIRepository;
+  }
+
+  @Override
+  public DebtPositionDTO createDebtPosition(DebtPositionDTO debtPositionDTO, Boolean massive, Boolean pagopaPayment, String accessToken, String operatorExternalUserId) {
+    log.info("Creating a DebtPosition having organizationId {}, debtPositionTypeOrgId {}, iupdOrg {}, ingestionFlowFileId {}, rowNumber {}", debtPositionDTO.getOrganizationId(),
+      debtPositionDTO.getDebtPositionTypeOrgId(), debtPositionDTO.getIupdOrg(), debtPositionDTO.getIngestionFlowFileId(), debtPositionDTO.getIngestionFlowFileLineNumber());
+
+    authorizeOperatorOnDebtPositionTypeService.authorize(debtPositionDTO.getOrganizationId(), debtPositionDTO.getDebtPositionTypeOrgId(), operatorExternalUserId);
+    validateDebtPositionService.validate(debtPositionDTO, accessToken);
+    verifyInstallmentUniqueness(debtPositionDTO);
+
+    if (Boolean.TRUE.equals(pagopaPayment)) {
+      debtPositionDTO.getPaymentOptions().stream()
+        .flatMap(po -> po.getInstallments().stream())
+        .forEach(installment -> {
+          Long orgId = debtPositionDTO.getOrganizationId();
+          String generatedIuv = generateIuvService.generateIuv(String.valueOf(orgId), accessToken);
+          installment.setIuv(generatedIuv);
+        });
+    }
+
+    DebtPositionDTO savedDebtPosition = debtPositionService.saveDebtPosition(debtPositionDTO);
+
+    log.info("END Create DebtPosition...");
+    return savedDebtPosition;
+  }
+
+  private void verifyInstallmentUniqueness(DebtPositionDTO debtPositionDTO) {
+    debtPositionDTO.getPaymentOptions().stream()
+      .flatMap(po -> po.getInstallments().stream())
+      .forEach(installmentNoPII -> {
+        long countDuplicates = installmentNoPIIRepository.countExistingInstallments(debtPositionDTO.getOrganizationId(), installmentNoPII.getIud(), installmentNoPII.getIuv(), installmentNoPII.getNav());
+        if (countDuplicates > 0) {
+          log.error("Duplicate record found for Installment with id {}", installmentNoPII.getInstallmentId());
```

Comment:
the requesting installment is not stored, thus it has not an id to print!
```suggestion
          log.error("Duplicate installments found for input Installment having IUD {}, IUV {}, NAV {} on organization {}", ...);
```